code/game/area/areas.dm

```diff
@@ -548,3 +555,131 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 	if(name == initial(name))
 		return name
 	return "[name] ([initial(name)])"
+
+// Day night stuff
+/**
+ * This will calculate all adjacent turfs in this area and add them to a cache for lighting effects.
+ *
+ * WARNING: This proc is VERY expensive and should be used sparingly.
+ */
+/area/proc/initialize_day_night_adjacent_turfs()
+	LAZYCLEARLIST(adjacent_day_night_turf_cache)
+	LAZYINITLIST(adjacent_day_night_turf_cache)
+
+	for(var/turf/iterating_turf in contents)
```

Comment:
Dunno if it's showing for you but GitHub is telling me I changed the indenting even though I didn't 🤷‍♀️

Reply:
worth noting that this is liable to force a good amount of list work, depending on the size of the area and how much has been drawn over it, since it resolves to be removed turfs.
it doesn't mean it's worse then the turfs in world method, but it can be liable to show up as a painpoint if called too early after init. how bad depends on the area, space is the classical worst case scenario.

Reply:
Done this, and tested. I haven't noticed any abnormalities?